lib/components/ModemUpdateDialogView.jsx

```diff
@@ -0,0 +1,150 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+    Button,
+    ControlLabel,
+    FormGroup,
+    Modal,
+    ModalBody,
+    ModalFooter,
+    ModalHeader,
+    ModalTitle,
+    Alert,
+} from 'react-bootstrap';
+
+
+export default class ModemUpdateDialogView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.onOk = props.onOk;
+        this.onWriteStart = this.onWriteStart.bind(this);
+    }
+
+    onWriteStart() {
+        this.setState({ timer: 0 });
+        if (this.intervalId) {
+            clearInterval(this.intervalId);
+        }
+        this.intervalId = setInterval(() => {
+            this.setState({ timer: this.state.timer + 1 });
+        }, 1000);
+        this.onOk();
+    }
+
+    render() {
+        const {
+            isVisible,
+            isWriting,
+            isWritingSucceed,
+            isWritingFail,
+            modemFwName,
+            progressMsg,
+            onCancel,
+        } = this.props;
+        if (this.intervalId && (isWritingSucceed || isWritingFail)) {
+            clearInterval(this.intervalId);
+        }
+        return (
+            <Modal show={isVisible} onHide={this.onCancel} backdrop={'static'}>
+                <ModalHeader>
+                    <ModalTitle className="modem-dialog-title">Modem DFU</ModalTitle>
+                    {isWriting &&
+                        <span className="glyphicon glyphicon-refresh glyphicon-spin" />
+                    }
+                </ModalHeader>
+                <ModalBody>
+                    <FormGroup>
+                        <ControlLabel>Modem firmware</ControlLabel>
+                        <div>{ modemFwName }</div>
+                    </FormGroup>
+                    <FormGroup>
+                        <ControlLabel>Process</ControlLabel>
+                        <div>{ progressMsg }</div>
+                    </FormGroup>
+                    <FormGroup>
+                        {isWritingSucceed &&
+                            <Alert bsStyle="success">
+                                Complete  successfully in {this.state.timer} seconds.
```

Comment:
Should be s/complete/completed/ ?
And seems to have one space too many.

Reply:
Fixed.